### PR TITLE
Add Time Range Filter for Bulk API

### PR DESCRIPTION
**Pull request by khansaad**
## Description

This PR adds the feature to get the bulk API result within a specified duration i.e. start and end time.

**Note: This is on top of [1358](https://github.com/kruize/autotune/pull/1358) which needs to be merged first.**

Fixes # (issue)

### Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Docs update
- [ ] Breaking change (What changes might users need to make in their application due to this PR?)
- [ ] Requires DB changes

## How has this been tested?

Please describe the tests that were run to verify your changes and steps to reproduce. Please specify any test configuration required. 

- [ ] New Test X
- [x] Functional testsuite

**Test Configuration**
* Kubernetes clusters tested on: **Minkube and Openshift**

## Checklist :dart:

- [x] Followed coding guidelines
- [x] Comments added
- [x] Dependent changes merged
- [ ] Documentation updated
- [ ] Tests added or updated

## Additional information

Include any additional information such as links, test results, screenshots here

### CODE REVIEW

File: src/main/java/com/autotune/common/datasource/DataSourceCollection.java [hunk 1]
```diff
@@ -144,6 +144,10 @@ public void addDataSourcesFromConfigFile(String configFileName) {
                     DataSourceInfo dataSourceInfo = new ExperimentDBService().loadDataSourceFromDBByName(name);
                     if (null != dataSourceInfo) {
                         LOGGER.error("Datasource: {} already exists!", name);
+                        // add the auth details to local object
```
Comment (msvinaykumar) [inline src/main/java/com/autotune/common/datasource/DataSourceCollection.java:147]: Is this change  related to PR ?

**Comment by dinogun** [comment]
@khansaad Please resolve the conflicts

File: src/main/java/com/autotune/common/datasource/DataSourceCollection.java (hunk 1 above)
Comment (khansaad) [reply src/main/java/com/autotune/common/datasource/DataSourceCollection.java:147]: No, this was pulled from #1358 . It needs to be merged first.

**Comment by dinogun** [comment]
@khansaad  This needs an example to be added in the Documentation

File: design/BulkAPI.md [hunk 2]
```diff
@@ -216,21 +256,21 @@ resource optimization in Kubernetes environments. Below is a breakdown of the JS
 
     - Each object in the `experiments` array has the following structure:
 
-  | Field                   | Type         | Description                                                              |
-              |-------------------------|--------------|--------------------------------------------------------------------------|
-  | `name`                  | `string`     | Name of the experiment, typically indicating a service name and deployment context. |
-  | `notification`          | `object`     | Notifications specific to this experiment (if any).                      |
-  | `recommendation`        | `object`     | Recommendation status and notifications specific to this experiment.     |
+  | Field            | Type     | Description                                                                         |
+              |------------------|----------|-------------------------------------------------------------------------------------|
```
Comment (dinogun) [inline design/BulkAPI.md:260]: Looks like the space at the beginning is causing it get messed up

File: design/BulkAPI.md [hunk 3]
```diff
@@ -216,21 +256,21 @@ resource optimization in Kubernetes environments. Below is a breakdown of the JS
...
 
   The `recommendation` field within each experiment provides information about recommendation processing status and
   errors (if any).
 
-  | Field                   | Type         | Description                                                              |
-              |-------------------------|--------------|--------------------------------------------------------------------------|
-  | `status`                | `string`     | Status of the recommendation (e.g., `"unprocessed"`, `"processed"`, `"processing"`, `"failed"`). |
-  | `notification`          | `object`     | Notifications related to recommendation processing.                      |
+  | Field          | Type     | Description                                                                                      |
+              |----------------|----------|--------------------------------------------------------------------------------------------------|
```
Comment (dinogun) [inline design/BulkAPI.md:271]: Same here

File: design/BulkAPI.md [hunk 4]
```diff
@@ -82,6 +82,46 @@ progress of the job.
...
+      "labels": {
+        "key1": "value1",
+        "key2": "value2"
+      }
+    }
+  },
+  "time_range": {
+    "start": "2024-11-01T00:00:00.000Z",
+    "end": "2024-11-15T23:59:59.000Z"
+  },
```
Comment (dinogun) [inline design/BulkAPI.md:111]: I'd suggest to add this into the JSON at the top, so there is a single JSON for reference. However you can add some explanation in this section

Comment (khansaad) [reply design/BulkAPI.md]: Updated now

File: design/BulkAPI.md [hunk 5]
```diff
@@ -216,28 +237,28 @@ resource optimization in Kubernetes environments. Below is a breakdown of the JS
...
-              |-------------------------|--------------|--------------------------------------------------------------------------|
-  | `name`                  | `string`     | Name of the experiment, typically indicating a service name and deployment context. |
-  | `notification`          | `object`     | Notifications specific to this experiment (if any).                      |
-  | `recommendation`        | `object`     | Recommendation status and notifications specific to this experiment.     |
+  | Field            | Type     | Description                                                                         |
+  |------------------|----------|-------------------------------------------------------------------------------------|
+  | `name`           | `string` | Name of the experiment, typically indicating a service name and deployment context. |
+  | `notification`   | `object` | Notifications specific to this experiment (if any).                                 |
+  | `recommendation` | `object` | Recommendation status and notifications specific to this experiment.                |
```
Comment (dinogun) [inline design/BulkAPI.md:244]: Can you please change this to `notifications` and `recommendations` everywhere, thanks

Comment (khansaad) [reply design/BulkAPI.md]: Done

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**